### PR TITLE
docs: update Grid docs

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

This PR will update docs for `Grid`.

#### Description of how you validated changes

Screen Shot 2022-05-13 at 4 28 07 PM

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/components/grid/react.mdx [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
-import { Example } from '@/components/Example';
+import {
+  DefaultGridExample,
+  GridResponsiveExample,
+  GridRowAndColumnSpanExample,
+} from './examples';
+import { Example, ExampleCode } from '@/components/Example';
 import { ComponentStyleDisplay } from '@/components/ComponentStyleDisplay';
 
 Grid primitive provides a grid container with style `display: grid`. Any of our other primitives can be used grid item children. To learn how to use Grid CSS properties, see the following documentation:
```
Comment (joebuono) [inline docs/src/pages/[platform]/components/grid/react.mdx:10]: A few minor tweaks: "Grid provides a CSS Grid container with style `display: grid`. Any Amplify UI components can be used as grid item children. To learn how to use CSS Grid properties..."

File: docs/src/pages/[platform]/components/grid/demo.tsx [hunk 2]
```diff
@@ -1,17 +1,98 @@
...
+  GridItemPropControls,
+  GridItemPropControlsProps,
+} from './GridItemPropControls';
+import { useGridItemProps } from './useGridItemProps';
+import { useGridContainerProps } from './useGridContainerProps';
+
+const propsToHeaderGridItemCode = (props: GridItemPropControlsProps) => {
+  return (
+    '<Card' +
+    (props.columnStart ? `\n  columnStart={${props.columnStart}}` : '') +
```
Comment (joebuono) [inline docs/src/pages/[platform]/components/grid/demo.tsx:25]: Just a heads up that we have a `getPropString` function in the `utils` folder that does this for us

**Review by joebuono — COMMENTED** [review]
Thank you so much for doing this refactor! 🙌 

I just had a few small comments tweaking some of the phrasing.

File: docs/src/data/links.ts [hunk 3]
```diff
@@ -219,7 +219,7 @@ export const layoutComponents = [
   {
     href: '/components/grid',
     label: 'Grid',
-    body: ``,
+    body: 'Grid primitive provides a grid container. Any of other primitives can be used as grid item children.',
```
[the remark this answers is not in the document]
Comment (zchenwei) [reply docs/src/data/links.ts:222]: Good idea

File: docs/src/pages/[platform]/components/grid/demo.tsx [hunk 4]
```diff
@@ -1,17 +1,98 @@
...
+    (props.columnEnd ? `\n  columnEnd={${props.columnEnd}}` : '') +
+    (props.rowStart ? `\n  rowStart={${props.rowStart}}` : '') +
+    (props.rowEnd ? `\n  rowEnd={${props.rowEnd}}` : '') +
+    '\n >' +
+    '\n  Header' +
+    '\n </Card>'
+  );
+};
+
+const propsToNavGridItemPcode = (props: GridItemPropControlsProps) => {
```
[the remark this answers is not in the document]
Comment (zchenwei) [reply docs/src/pages/[platform]/components/grid/demo.tsx:35]: Good catch 🤣

File: docs/src/pages/[platform]/components/grid/demo.tsx [hunk 5]
```diff
@@ -54,105 +135,146 @@ const defaultMainGridItemStyleProps: GridItemStyleProps = {
...
+      ...defaultFooterGridItemStyleProps,
+      name: 'GridItemFooter',
+    }
   );
-  const headerGridItemProps = useGridItemProps(defaultHeaderGridItemStyleProps);
-  const footerGridItemProps = useGridItemProps(defaultFooterGridItemStyleProps);
-  const navGridItemProps = useGridItemProps(defaultNavGridItemStyleProps);
-  const mainGridItemProps = useGridItemProps(defaultMainGridItemStyleProps);
+  const navGridItemProps = useGridItemProps(
+    demoState.get('GirdItemNav') || {
```
[the remark this answers is not in the document]
Comment (zchenwei) [reply docs/src/pages/[platform]/components/grid/demo.tsx:153]: Good catch!

File: docs/src/pages/[platform]/components/grid/demo.tsx (hunk 2 above)
Comment (zchenwei) [reply docs/src/pages/[platform]/components/grid/demo.tsx:25]: ohhh, I like this util.

File: docs/src/pages/[platform]/components/utils/getPropString.ts [hunk 6]
```diff
@@ -1,2 +1,13 @@
-export const getPropString = (prop: string | number, propName: string) =>
-  prop ? `\n  ${propName}="${prop}"` : '';
+import { ResponsiveStyle } from '@aws-amplify/ui-react';
+
+export const getPropString = (
+  prop: ResponsiveStyle<any>,
+  propName: string,
+  indents: number = 2
```
Comment (reesscot) [inline docs/src/pages/[platform]/components/utils/getPropString.ts:6]: nice feature!

**Review by joebuono — APPROVED** [review]
LGTM!

**Pull request merged**